orchestrator-salt/src/main/resources/salt/salt/metadata/init.sls

```diff
@@ -9,4 +9,11 @@ cluster_metadata:
     - name: /opt/metadata/cluster.json
     - dataset:
         {{ metadata }}
+    - formatter: json
+
+node_attributes:
```

Comment:
What happens if the `hostattrs` are empty? As part of an upgrade we upgrade the salt states and pillar as well, but they won't have any attributes.

Reply:
https://jira.cloudera.com/browse/CB-8147?focusedCommentId=4095583&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4095583

has the output in case there are no attributes defined. This isn't specifically for upgrades, but I'd expect an upgraded CB with this change to exercise the same code path and write out empty attributes.